sys/include/sema.h

```diff
@@ -179,6 +164,49 @@ static inline int sema_try_wait(sema_t *sema)
     return _sema_wait(sema, 0, 0);
 }
 
+#if IS_USED(MODULE_XTIMER)
+/**
+ * @brief   Wait for a semaphore being posted.
+ *
+ * @pre `(sema != NULL)`
+ *
+ * @param[in]  sema     A semaphore.
+ * @param[in]  timeout  Time in microseconds until the semaphore times out.
+ *                      0 does not wait.
+ *
+ * @return  0 on success
+ * @return  -ETIMEDOUT, if the semaphore times out.
+ * @return  -ECANCELED, if the semaphore was destroyed.
+ * @return  -EAGAIN,    if the semaphore is not posted (only if timeout = 0)
+ */
+static inline int sema_wait_timed(sema_t *sema, uint64_t timeout)
+{
+    return _sema_wait(sema, (timeout != 0), timeout);
+}
+#endif
+
+#if IS_USED(MODULE_ZTIMER)
+/**
+ * @brief   Wait for a semaphore being posted, using ztimer as backend
+ *
+ * @pre `(sema != NULL)`
+ * @pre `(clock != NULL)`
+ *
+ * @param[in]  sema     A semaphore.
+ * @param[in]  clock    ztimer clock to use
+ * @param[in]  timeout  Time in microseconds until the semaphore times out.
+ *                      0 does not wait.
+ *
+ * @return  0 on success
+ * @return  -ETIMEDOUT, if the semaphore times out.
+ * @return  -ECANCELED, if the semaphore was destroyed.
+ * @return  -EAGAIN,    if the semaphore is not posted (only if timeout = 0)
+ */
+static inline int sema_wait_timed_ztimer(sema_t *sema,
```

Comment:
How about also adding:

```
static inline int sema_wait_timed(sema_t *sema, uint64_t timeout)
{
    return _sema_wait_ztimer(sema,  (timeout != 0), ZTIMER_USEC, timeout);
}
```

when ztimer is used?

Reply:
How the exact call is made should be sorted after factoring a bit the C implementation, but I think having this wrapper to keep the same API would make sense.

Reply:
I don't really like it, as this adds quite some complexity regarding resolution of actual implementation based on time backends included in the build, as we allow builds with ztimer and xtimer simultaneously. So I think sticking with a strict separation of `sema_wait_timed() -> xtimer` and `sema_wait_timed_ztimer() -> ztimer` in the API is the cleanest solutiion. If (one day) xtimer is deprecated, we can easily introduce the mapping proposed above for backward compatibility then...

Reply:
So can you add that wrapper if timer is used? Also, does it make sense to have calls that use `ztimer` and `xtimer`?

Reply:
I had missed https://github.com/RIOT-OS/RIOT/pull/15782#discussion_r604705742, then never mind this comment.